test/test_torch.py

```diff
@@ -2463,6 +2463,11 @@ def test_flatten(self):
                 with self.assertRaisesRegex(RuntimeError, 'start_dim cannot come after end_dim'):
                     src.flatten(2, 0)
 
+        def test_unflatten(self):
+            self.assertEqual(torch.tensor([]).unflatten(0, (0, 1, 0)), torch.empty(0, 1, 0))
+            self.assertEqual(torch.tensor([1]).unflatten(0, (1, 1)), torch.tensor([[1]]))
+            self.assertEqual(torch.tensor([[1, 2, 3, 4]]).unflatten(1, (2, 2)), torch.tensor([[[1, 2], [3, 4]]]))
+
```

Comment:
Could we also test the following:
- pass in a torch.Size for `sizes` (since we advertise we support that)
- pass in a List[int] for `sizes` (all of our functions that take torch.Size also support List[int])
- pass in an empty list for sizes (what happens in this case?)
- check the error message when passing in an invalid size (e.g., `torch.randn(3).unflatten([1, 1])`)
- call unflatten on a scalar tensor (`torch.tensor(1).unflatten(0, ??)`)

To test error messages one can use assertRaisesRegex. Example: https://github.com/pytorch/pytorch/blob/b85216887ba76dc7e1e0035c77ebe358ac48f90d/test/test_torch.py#L123-L124